### PR TITLE
embed font files with //go:embed

### PR DESCRIPTION
**Pull request by flibustenet**
For issue go-pdf/fpdf#13 
I just added the same files that was embedded in `embedded.go`. I mean i didn't add others `*.map` and `*.json` of `font` directory to keep exactly the same functionalities for now.

### CODE REVIEW

File: embedded.go [hunk 1]
```diff
@@ -19,541 +19,25 @@ package fpdf
...
-}
+//go:embed font_embed/*.map
+var embeddedFontMap embed.FS // for util.go (f *Fpdf) UnicodeTranslatorFromDescriptor
+
+//go:embed font_embed/*.json
+var embeddedFontJson embed.FS
 
-func (f *Fpdf) coreFontReader(familyStr, styleStr string) (r *strings.Reader) {
+func (f *Fpdf) coreFontReader(familyStr, styleStr string) (r io.Reader) {
```
Comment (sbinet) [inline embedded.go:33]: this should probably be returning an `io.ReadCloser`.
(and modify calling code to add a `defer o.Close()`)

File: util.go [hunk 2]
```diff
@@ -236,15 +236,13 @@ func UnicodeTranslatorFromFile(fileStr string) (f func(string) string, err error
...
-	var str string
-	var ok bool
 	if f.err == nil {
 		if len(cpStr) == 0 {
 			cpStr = "cp1252"
 		}
-		str, ok = embeddedMapList[cpStr]
-		if ok {
-			rep, f.err = UnicodeTranslator(strings.NewReader(str))
+		emb, err := embeddedFontMap.Open("font_embed/" + cpStr + ".map")
```
Comment (sbinet) [inline util.go:243]:
```suggestion
		emb, err := embeddedFontMap.Open(filepath.Join("font_embed", cpStr + ".map"))
```

File: util.go [hunk 3]
```diff
@@ -236,15 +236,13 @@ func UnicodeTranslatorFromFile(fileStr string) (f func(string) string, err error
...
+		if err == nil {
+			rep, f.err = UnicodeTranslator(emb)
```
Comment (sbinet) [inline util.go:245]:
```suggestion
			defer emb.Close()
			rep, f.err = UnicodeTranslator(emb)
```

File: embedded.go [hunk 4]
```diff
@@ -19,541 +19,25 @@ package fpdf
...
 	key := familyStr + styleStr
-	str, ok := embeddedFontList[key]
-	if ok {
-		r = strings.NewReader(str)
+	key = strings.ToLower(key)
+	emb, err := embeddedFontJson.Open("font_embed/" + key + ".json")
```
Comment (sbinet) [inline embedded.go:36]:
```suggestion
	emb, err := embeddedFontJson.Open(filepath.Join("font_embed", key + ".json"))
```

File: embedded.go [hunk 5]
```diff
@@ -19,541 +19,25 @@ package fpdf
...
-	"zapfdingbats": `{"Tp":"Core","Name":"ZapfDingbats","Up":-100,"Ut":50,"Cw":[0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,278,974,961,974,980,719,789,790,791,690,960,939,549,855,911,933,911,945,974,755,846,762,761,571,677,763,760,759,754,494,552,537,577,692,786,788,788,790,793,794,816,823,789,841,823,833,816,831,923,744,723,749,790,792,695,776,768,792,759,707,708,682,701,826,815,789,789,707,687,696,689,786,787,713,791,785,791,873,761,762,762,759,759,892,892,788,784,438,138,277,415,392,392,668,668,0,390,390,317,317,276,276,509,509,410,410,234,234,334,334,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,732,544,544,910,667,760,760,776,595,694,626,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,788,894,838,1016,458,748,924,748,918,927,928,928,834,873,828,924,924,917,930,931,463,883,836,836,867,867,696,696,874,0,874,760,946,771,865,771,888,967,888,831,873,927,970,918,0]}`,
-}
+//go:embed font_embed/*.map
+var embeddedFontMap embed.FS // for util.go (f *Fpdf) UnicodeTranslatorFromDescriptor
+
+//go:embed font_embed/*.json
+var embeddedFontJson embed.FS
```
Comment (sbinet) [inline embedded.go:31]: is there really a need for 2 different `embed.FS`?

I'd just slam everything together.
```suggestion
//go:embed font_embed/*.json
var embFS embed.FS
```

**Review by sbinet — COMMENTED** [review]
thanks for the PR.

I have a couple of comments and suggestions.
see below.

File: util.go (hunk 2 above)
Comment (flibustenet) [reply util.go:243]: The separator `/` is hardcoded in `embed.FS`
> The path separator is a forward slash, even on // Windows systems.

File: embedded.go (hunk 5 above)
Comment (flibustenet) [reply embedded.go:31]: At first i put embeddedFontMap in util.go, it's why it was separated. But it's better in the same place and one FS.

File: embedded.go (hunk 1 above)
Comment (flibustenet) [reply embedded.go:33]: `Close` doesn't do anything in embed.FS
`func (f *openFile) Close() error               { return nil }`

Comment (sbinet) [reply embedded.go:30]: sure, but that's kind of an implementation detail.
`io/fs.File.Close` makes no promise, except from following a "minimal" `os.File`-like "interface".

Comment (flibustenet) [reply embedded.go]: That make sense, it's safer.

**Review by sbinet — APPROVED** [review]
thanks.

**Pull request merged**